### DISCUSSION TITLE
Hochladen von .geojson files schlägt fehl

### DISCUSSION THREAD

**Discussion by ue-black** [e-netze/webgis-community#105] [General]
Hallo!

Wir haben die Anfrage Redlining Geometrien von WebGIS 4 Projekten ins WebGIS 5 zu übernehmen. Folgende Vorgehensweise halten wir für schlau (Tipps erwünscht):
- Export des Redlining als Shape
- Öffnen im QGIS und exportieren als .geojson
- Hochladen des .geojson im WebGIS 5

Dabei sind wir auf ein paar Hürden gestoßen:

1. Das .geojson File funktioniert nicht beim Upload.

![image](https://github.com/gis-eni/webgis-forum/assets/127403099/55b8e03e-07ae-4eee-ba06-d6d5490b75a8)

Ist grundsätzlich egal, weil man kann ja das .geojson in ein .json file umwandeln. Dann funktioniert es. Muss man wissen.

2. Bei dem .json handelt es sich um Points. Dabei bekomme ich immer folgende Fehlermeldung:

![image](https://github.com/gis-eni/webgis-forum/assets/127403099/c63ada72-5b7f-42e0-b899-17fd11a528ea)

Hier das file: [testwgs84.json](https://drive.cnv.at/s/bccKYbSr5G2oHyj) Das Format des JSON sollte mMn korrekt sein.

Ich habe zu Testzwecken die ersten zwei Punkte des obigen Files händisch zu einem LineString umgebaut (und die restlichen Punkte gelöscht). Syntax ist gleich nur statt Point habe ich jetzt LineString. Danach funktioniert der Import problemlos. Hier das LineString JSON: [testwgs84_linestring](https://drive.cnv.at/s/4NzeQDdxsDqGz5q)

Warum kann ich keine Punkte importieren? Bitte um Hilfe :-)

**Comment by gis-eni** [comment]
Hallo,

Das Hochladen von Redlining Projekten entspricht eigentlich nicht einem GeoJson Import. Hier wird eigentlich ein Redling Projekt erwartet. Das ist zwar streng genommen ein GeoJSON File, allerdings mit ein zusätzlichen *Properties* wie ``_meta``.

Dh. Du kannst ein GeoJSON aus dem Redlining in anderen Applikationen laden, aber nicht zwingend umgekehrt.
Im ``_meta``  Tag stehen Dinge wir die Farbe, das Werkzeug mit dem der Punkt gezeichnet wurde usw.

Du kannst probieren, die ein File von WebGIS 5 herunterzuladen, dann siehst du, welche Informationen im ``_meta`` Tag drinnen stehen. Genau so müsstest du dann dein GeoJSON anpassen, damit es auch wieder aufgeht.

Das ist natürlich umständlich.

Man könnte das so umbauen, dass man auch normale "geojson" Dateien hochladen kann, und dann für die Metadaten default Werte verwendet werden. Dann gehen halt mache Dinge wie Farben, Texte usw verloren.

Ob das aber der richtige Weg ist um Redlining von alt auf Neu zu migrieren stelle ich in Frage. Die Zeichenwerkzeuge von 4 und 5 sind nicht kompatibel. Da geh immer was verloren.

Grundsätzlich: Redlining ist eigentlich etwas temporäres. Wozu eine Migration von WebGIS 4 auf WebGIS 5?
Wenn es wirklich Projekte gibt, die so wichtig sind, dass sie länger aufbehalten werden müssen, stellt sich die Frage, ob hier Redling das richtige Werkzeug ist. Vielleicht ist für diese Anwendungsfälle eine Editierumgebung die bessere Wahl. Da kommen die Objekte gleich in die Datenbank und gehen nicht mehr verloren. 
Die Migration würde dann bedeuten: WebGIS Redlining => Shape download => Migration in die Datebank (postgis) => Dienst zum Editieren erzeugen (gView),

**Reply by ue-black** [reply]
Hallo Jürgen,

da bin ich ganz bei dir, Redlining sollte für temporäre Zwecke verwendet werden und für langfristige Dinge sollte man den von dir oben beschriebenen Weg (oder ein QGIS Projekt, wenn der User etwas GIS KnowHow hat). Es werden wahrscheinlich noch ein paar User:innen kommen die riesige Redlining Projekte haben und da wünschen sich diese dann eine Lösung von uns. Im November drehen wir den 4er endgültig ab. 

Der Vorschlag mit dem Anpassen des Imports mit Standard _meta Werten ist sicher überlegenswert, ich bespreche das dann nächste Woche mit "meinem" Jürgen, wenn er wieder zurück vom Urlaub ist. Wir melden uns, Danke schon mal!

**Reply by marko-jer** [reply]
Hallo Klaus,
ich habe seinerzeit in py-script geschrieben, das die Umwandlung einigermaßen verlustfrei umsetzt und über den ProcessingServer zugängig gemacht. Kannst es ausprobieren -- [https://gis.stmk.gv.at/ps/?appId=redline_new](https://gis.stmk.gv.at/ps/?appId=redline_new) 
Vielleicht funktioniert es noch. ;) (Das Koordinatensystem des shp könnte noch ein Problem darstellen)

**Comment by ue-black** [comment]
Ich habe Markos Lösung getestet und sie funktioniert bestens für unsere Anforderungen. MMn kann man diese Diskussion schließen.
